### PR TITLE
Users can edit their articles

### PR DESCRIPTION
**Pull request by thesuss**
Piv:
https://www.pivotaltracker.com/story/show/130580783

Users can edith theirs, and not someone elses article. 
Visitors can not edit articles.

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -20,7 +20,25 @@ def create
       redirect_to article_path(@article)
     else
       redirect_to new_user_session_path
-      flash[:notice] = error_message
+      flash[:alert] = error_message
```
Comment (tochman) [inline app/controllers/articles_controller.rb:22]: I would suggest setting the flash before the redirect. Also what is `error_message`? Where is it defined?

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -20,7 +20,25 @@ def create
...
+    end
+  end
+
+  def edit
+    if current_user
+    else
```
Comment (tochman) [inline app/controllers/articles_controller.rb:30]: so `if current_user` then what?

File: app/controllers/articles_controller.rb [hunk 3]
```diff
@@ -20,7 +20,25 @@ def create
...
+      redirect_to new_user_session_path
+      flash[:alert] = error_message
```
Comment (tochman) [inline app/controllers/articles_controller.rb:31]: Same thing here, set flash before redirecting.

File: app/controllers/articles_controller.rb [hunk 4]
```diff
@@ -20,7 +20,25 @@ def create
...
+    end
+  end
+
+  def update
+    @article = Article.find(params[:id])
+    if @article.update(article_params)
+      redirect_to @article
```
Comment (tochman) [inline app/controllers/articles_controller.rb]: is this `redirect_to` actually working? What path is being requested?

File: app/controllers/articles_controller.rb [hunk 5]
```diff
@@ -20,7 +20,25 @@ def create
...
+    else
+      flash[:alert] = edit_error_message
+      render 'edit'
```
Comment (tochman) [inline app/controllers/articles_controller.rb]: I would suggest using a symbol. feels and looks cleaner

```
render :edit
```

File: app/controllers/articles_controller.rb [hunk 6]
```diff
@@ -33,8 +51,10 @@ def index
   def article_params
     params.require(:article).permit(:title, :body, :user_id, :tag_list)
   end
-
   def error_message
     "You are not logged in - you need to be logged in to see the page you were trying to reach"
   end
```
Comment (tochman) [inline app/controllers/articles_controller.rb:57]: So this is a kind of authorization method. I understand this. I was under the impression that you guys ware about to implement CanCanCan as an authorization module.  I think you need to discuss this. If you decide NOT to implement CanCanCan, I have no issues with merging this. But I want you to make an argument for implementing authorizations this way.

File: app/views/articles/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<% @article = Article.find_by_id(params[:id])%>
```
Comment (tochman) [inline app/views/articles/edit.html.erb]: This kind of calls are not okay to do in a view. We have controllers for that. ;-)

File: app/views/articles/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+<h1 class="content-head is-center">Update article</h1>
+<%= form_for @article, url:{action: "update"}, html: { class: 'pure-form pure-form-aligned' } do |f| %>
```
Comment (tochman) [inline app/views/articles/edit.html.erb:2]: Again, `url:{action: "update"}, html: { class: 'pure-form pure-form-aligned' }` feels a little to much and a bit of mix of syntax used for `form_for` and `form_tag`  There must be a cleaner way to do this.

File: app/views/articles/new.html.erb [hunk 9]
```diff
@@ -16,17 +16,17 @@
...
+    <%= f.label :body %>
     <%= f.text_area :body %>
   </div>
 
   <div class="pure-control-group">
-    <%= f.label :tag_list, 'Tags' %><br>
+    <%= f.label :tag_list, 'Tags' %>
     <%= f.text_field :tag_list, placeholder: "Tags separated with comma" %>
   </div>
   <p>
```
Comment (tochman) [inline app/views/articles/new.html.erb:32]: There seem to be a lot of repetition of code on `new` and `edit` Can you try to DRY this out a bit`

File: app/views/articles/show.html.erb [hunk 10]
```diff
@@ -11,3 +11,7 @@
     <%= link_to tag, tag_path(tag) %><% unless tag == @last || @tags.nil? %>,<% end %>
   <% end %>
 </p>
+<% if current_user == @article.user%>
```
Comment (tochman) [inline app/views/articles/show.html.erb:14]: Again, this works but an authorization module would do a better job.

File: app/views/articles/show.html.erb [hunk 11]
```diff
@@ -11,3 +11,7 @@
...
+  <%= link_to 'Edit article', edit_article_path %>
+<% else %>
```
Comment (tochman) [inline app/views/articles/show.html.erb]: No need to include empty `else` blocks.

File: features/articles.feature [hunk 12]
```diff
@@ -6,13 +6,19 @@ Background:
   Given the following users exist:
   | name   | email           | password  |  password_confirmation |
   | Anna   | anna@random.com | password  |  password              |
+  | Boon   | boon@random.com | password  |  password              |
```
Comment (tochman) [inline features/articles.feature]: Not sure you need `password_confirmation` when creating users with FactoryGirl. The less code we have, the better.

**Review by tochman — COMMENTED** [review]
A few comments

File: features/articles.feature (hunk 12 above)
Comment (thesuss) [reply features/articles.feature]: Fixed

File: app/views/articles/show.html.erb (hunk 11 above)
Comment (thesuss) [reply app/views/articles/show.html.erb]: Fixed

File: app/views/articles/edit.html.erb (hunk 7 above)
Comment (thesuss) [reply app/views/articles/edit.html.erb]: Fixed

File: app/controllers/articles_controller.rb (hunk 5 above)
Comment (thesuss) [reply app/controllers/articles_controller.rb]: Fixed

File: app/controllers/articles_controller.rb (hunk 4 above)
Comment (thesuss) [reply app/controllers/articles_controller.rb]: Fixed

File: app/controllers/articles_controller.rb (hunk 3 above)
Comment (thesuss) [reply app/controllers/articles_controller.rb:31]: Fixed

File: app/controllers/articles_controller.rb (hunk 2 above)
Comment (thesuss) [reply app/controllers/articles_controller.rb:30]: I think I fixed.

**Pull request merged**